### ISSUE TITLE
Validation accuracy varies widely

### ISSUE THREAD

**Issue by nahidalam** [RasaHQ/rasa_core#616]
**Rasa Core version**: 0.9.0a6

**Python version**: 3.6

**Operating system** (windows, osx, ...): OSx

**Issue**:
While training the dialogue management model, I have noticed the validation accuracy varies widely. Lost likely the model is overfitting as I have a really small amount of data. I am in the early stage of development so more data will come later. Any suggestion on improving this without adding more data?

**Content of domain file** (if used & relevant):
```
slots:
  product:
    type: text
  router:
    type: text


intents:
 - greet
 - goodbye
 - order_product


entities:
 - product
 - router

templates:
  utter_greet:
    - 'Hello! My name is Tom. How can I help you today?'
  utter_goodbye:
    - 'Talk to you later.'
    - 'Bye'
  utter_affirm:
    - 'Thats right'
    - 'You are correct'
  utter_ask_product_name:
    - 'Which product?'
  utter_default:
    - "How can I help you today?"


actions:
 - utter_greet
 - utter_goodbye
 - utter_affirm
 - utter_default 
 - utter_ask_product_name
 - actions.ActionOrderProduct

```

**Comment by dipalira** [comment]
Could you expand on what you mean by _varying_. Meanwhile you can do the following to help diagnose the possible problems - 
1. Changing the number of epochs
2. Adding more intents/actions to your bot. This would increase the overall complexity of the model.
3. Try to regularize/ change hyper-parameters of the architecture
Adding data points would definitely reduce the fluctuations!

**Comment by nahidalam** [comment]
Thanks @dipalira  for the suggestion. What I mean by `varies widely` is if I train multiple times, I see the validation accuracy ranging from 40% to 80%. Couple times I also saw 100% so thats definitely overfitting.

**Comment by dipalira** [comment]
@nahidalam My bot shows similar variations. But there is no problem in it's functioning with large or small amount of dataset till now! It would be interesting to see what everyone else has to say.
